diffgraph/graph_manager.py

```
@@ -142,48 +162,58 @@ def get_connected_components(self, start_component: str, max_depth: int = 3) ->
+            mermaid.append('    end')
 
         # Add edges between components
         for source, target in self.component_graph.edges():
-            mermaid.append(f'    {source.replace("/", "_").replace("::", "_")} --> {target.replace("/", "_").replace("::", "_")}')
-
-        # Add style definitions
-        mermaid.append("    classDef change_added fill:green,stroke:#333,stroke-width:2px")
-        mermaid.append("    classDef change_deleted fill:red,stroke:#333,stroke-width:2px")
-        mermaid.append("    classDef change_modified fill:orange,stroke:#333,stroke-width:2px")
-        mermaid.append("    classDef change_unchanged fill:gray,stroke:#333,stroke-width:2px")
+            mermaid.append(f'    {re.sub(r'[^a-zA-Z0-9_]', '_', source)} --> {re.sub(r'[^a-zA-Z0-9_]', '_', target)}')
```

Pylint (3.3.7)

[error] Parsing failed: 'invalid decimal literal (diffgraph.graph_manager, line 200)'

(E0001)